### PR TITLE
Added pin screen stuff

### PR DESCRIPTION
**Pull request by Harrierr**
Please note none of this code has been tested in over 6 months, if someone could test it before merging it would be good.

### CODE REVIEW

**Comment by TheTS** [comment]
"Here is some random code I didn't test" 😂

**Comment by MerlijnWajer** [comment]
He said it worked last time he tested it, but that was some time ago. It takes time to ge tan account with a pin screen.

**Comment by Harrierr** [comment]
It was fully tested back on Olly-/SRL but I never got chance to pull it and I've only just re-based it against SRL/SRL. It /should/ work since I doubt any changes have happened to the pinscreen. 

Edit: Until/if FindColors and FindColorsTolerance is fixed on the linux build I cannot test this anyway.

File: osr/interfaces/bankscreen.simba [hunk 1]
```diff
@@ -651,7 +667,101 @@ begin
...
+    begin
+      Inc(F);
+      if (F > 2) then
+        srl.Writeln('Your pin is wrong!', dtFatal);
+      Self.EnterPin;
+    end;
+    if (GetTickCount64 > T) then
+      Exit(False);
+  end;
+  Result := Self.IsOpen(RandomRange(2000, 6000));
```
Comment (smithtrenton) [inline osr/interfaces/bankscreen.simba]: Shouldn't this be is not open? If I enter the pin correctly the bankscreen should be open, not the pinscreen(which is self.isopen)

File: osr/interfaces/bankscreen.simba [hunk 2]
```diff
@@ -651,7 +667,101 @@ begin
...
+
+  T := GetTickCount64 + RandomRange(15000, 20000);
+  while not BankScreen.IsOpen() do
+  begin
+    if ChatBox.ClickContinue then
+    begin
+      Inc(F);
+      if (F > 2) then
+        srl.Writeln('Your pin is wrong!', dtFatal);
+      Self.EnterPin;
```
Comment (smithtrenton) [inline osr/interfaces/bankscreen.simba]: Infinite recursive loop?

**Comment by TheTS** [comment]
All I'm saying is, these things tend to move through a whole lot quicker if you confirm they have been thoroughly tested when submitting. A few days waiting for a pin is negligible compared to how long this will sit here.

**Comment by Harrierr** [comment]
I just found some stuff in my git checkout and committed it, but somewhere along the line I hard rebased it so it deleted the last commit. I tested this 5 times and it opened fine but I haven't tested what happens under lag yet.

**Comment by slackydev** [comment]
Merged :)

**Pull request merged**